lib/backend/k8s/conversion/workload_endpoint.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2016-2020 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO move the WorkloadEndpoint converters to is own package. Some refactoring of the annotation and label constants
+// is necessary to avoid circular imports, which is why this has been deferred.
+package conversion
+
+import (
+	"net"
+
+	kapiv1 "k8s.io/api/core/v1"
+
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+)
+
+type WorkloadEndpointConverter interface {
+	VethNameForWorkload(namespace, podName string) string
+	InterfaceForPod(pod *kapiv1.Pod, iface string) (*PodInterface, error)
+	InterfacesForPod(pod *kapiv1.Pod) ([]*PodInterface, error)
+	DefaultInterfaceForPod(pod *kapiv1.Pod) (*PodInterface, error)
+	PodToWorkloadEndpoints(pod *kapiv1.Pod) ([]*model.KVPair, error)
+	PodToWorkloadEndpoint(pod *kapiv1.Pod, ifaceName string) (*model.KVPair, error)
+	PodToDefaultWorkloadEndpoint(pod *kapiv1.Pod) (*model.KVPair, error)
+}
+
+// PodInterface contains the network configuration for a particular interface on a Pod.
+type PodInterface struct {
+	IsDefaultInterface bool
+	NetworkName        string
+	HostSideIfaceName  string
+	InsidePodIfaceName string
+	InsidePodGW        net.IP
+	IPNets             []*cnet.IPNet
+}
+
+func GetWEPConverter() WorkloadEndpointConverter {
```

Comment:
Should probably call this `NewWorkloadEndpointConverter` to indicate it's creating a new struct, and to be more aligned with typical naming conventions. (we're not actually doing a "get" of any sort, it's just initializing a struct)

Reply:
Good call, I'll change it.